### PR TITLE
lib/epoch - getMembers

### PR DESCRIPTION
**Pull request by mixmix**
A function which loads the membership of an epoch (along with those who have been marked for exclusion.

### CODE REVIEW

**Comment by mixmix** [comment]
@Powersource this and the #83 PR are both failing because exclusion isn't quite working yet. If this is a surprise, we should def talk!!

File: lib/tangles/get-tangle-updates.js [hunk 1]
```diff
@@ -9,7 +9,16 @@ const B_CONTENT = Buffer.from('content')
...
 
-module.exports = function getUpdates(ssb, tangle, root, cb) {
+function getTangleUpdates(ssb, tangle, root, cb) {
+  pull(
+    tangleUpdateStream(ssb, tangle, root),
+    pull.collect(cb)
+  )
+}
+getTangleUpdates.stream = tangleUpdateStream
+module.exports = getTangleUpdates
```
Comment (mixmix) [inline lib/tangles/get-tangle-updates.js:19]: I just tweaked this file so you can get _EITHER_ a callback _OR_ a stream. :wink:

File: lib/epochs.js [hunk 2]
```diff
@@ -108,6 +117,8 @@ function reduceEpochs (ssb, groupId, cb) {
             })
           }),
           pull.collect((err, nodes) => {
+            if (err) return cb(err)
```
Comment (Powersource) [inline lib/epochs.js:120]: `clarify`ing errs would be nice

File: lib/tangles/get-tangle-updates.js (hunk 1 above)
Comment (Powersource) [reply lib/tangles/get-tangle-updates.js:19]: aah that's what the .stream is for i was confused

File: lib/epochs.js [hunk 3]
```diff
@@ -119,12 +130,32 @@ function reduceEpochs (ssb, groupId, cb) {
...
-function toFeedURI (id) {
-  if (id.startsWith('@')) return fromFeedSigil(id)
-  else return id
+
+function getEpochMembers (ssb, epochId, cb) {
+  const members = new Set()
+  const toExclude = new Set()
+
+  pull(
+    getTangleUpdates.stream(ssb, 'members', epochId),
```
Comment (Powersource) [inline lib/epochs.js:142]: aah epochId is the root msg id of the epoch? i guessed that it was the epoch secret. maybe call it epochRoot?

**Review by Powersource — CHANGES_REQUESTED** [review]
Looks nice just some small stuff

File: lib/epochs.js (hunk 3 above)
Comment (mixmix) [reply lib/epochs.js:142]: epochRoot is a good idea. In practice id === root for tangles, but this id is waaaaay overloaded as a term

**Pull request merged**